### ISSUE TITLE
2.0.0 : No option for video frame autofit if larger?

### ISSUE THREAD

**Issue by Question2** [clsid2/mpc-hc#2165]
Im running into a problem with some high res video files. If the res is too high, the video will not fit into the player unless i manually set the video frame to something like "touch window from inside". But once MPC is set to this mode, it will also upscale videos with a resolution smaller than the player. E.G. If i am playing a low res 640x480 video, it will upscale it to fit the player window size which is not ideal.

Does MPC not have an option to autofit the video frame if the video res is larger than the player? It would solve this problem easily. Kind of like how some image viewers have a "autofit only if image is larger" option.

**Comment by clsid2** [comment]
Disable fixed window size if you don't want to upscale.

**Comment by Question2** [comment]
> Disable fixed window size if you don't want to upscale.

Where is that option? Also, im talking specifically about the video frame, not the size of the MPC window.

**Comment by clsid2** [comment]
Options > player > remember last window size
Options > playback > auto-zoom

With default settings the player window will adjust automatically to the video size. So you are obviously not using defaults.

**Comment by Question2** [comment]
@clsid2 : Im not sure if we are on the same page here. I just checked and the player is on default settings, which is "remember last window size" and "auto-zoom 100%".

Im talking specifically about the video frame size, not the player window size. For example, if the video file has a res of 1920x1080, under default settings, part of the video will not be visible because things like the playback bar, the top menu bar, etc will cover part of the video. I can get around this by setting the video frame size to something smaller like "half normal size", but if i play another video with a smaller res, i would need to change this back to "normal size".

Is there no "auto fit" setting for the video frame specifically? So for example, if the video is of a large enough size (e.g. 1920x1080), then it will reduce the size of the video frame so that i can see the whole video without the playback bar, etc covering it, but it will not do this if the video has a small resolution (e.g. 640x480).

Example : 

![image](https://github.com/clsid2/mpc-hc/assets/1191378/ff6d0ba8-1191-4495-bb2c-20b19630310c)

![image](https://github.com/clsid2/mpc-hc/assets/1191378/f7de8e15-25ea-4346-a6b8-7c713a3e1107)

Basically, in the first screenshot, the video isnt covered by anything because the resolution is small enough. But in the second screenshot, part of the video is hidden and i have to manually lower the video frame size to compensate, but it is annoying having to manually switch between video frame sizes depending on the resolution of the video file. Is there a way to get MPC to reduce the video frame size ONLY if the resolution is too big?

**Comment by clsid2** [comment]
You are **not ** using default settings. "remember last window size" is not a default, nor is "normal size" video frame.

You can fix your problem by just not fucking maximizing the window. Then small video will be shown at exactly 100%. Watching big video? Go fullscreen.

**Comment by clsid2** [comment]
I am not going to add extra modes because people are doing stupid things.

I am not going to debate things with people who can't even follow simple instructions.
